editor/scene/connections_dialog.cpp

```diff
@@ -199,7 +199,10 @@ void ConnectDialog::_tree_node_selected() {
 		return;
 	}
 
-	dst_path = source->get_path_to(current);
+	if (!source->is_class("Resource")) {
+		dst_path = Object::cast_to<Node>(source)->get_path_to(current);
```

Comment:
```suggestion
	Node *source_node = Object::cast_to<Node>(source);
	if (source_node) {
		dst_path = source_node->get_path_to(current);
```
`is_class()` should be avoided.